rfcs/table.md

```diff
@@ -0,0 +1,321 @@
+# Table Component
+
+Represents control to render tabular data.
+
+## Features
+- Sorting
+- Basic mobile responsiveness (overflow scrollbar)
+
+
+## Unsupported
+- Pagination
+- Filtering
+- Fixed Header
+- Fixed Column
+- Striped Tables
+- Hoverable row styling
+- Multi-column sort
+- Row headers
+- Expandable Row
+
+## Usage
+
+### Basic usage
+#### Render static tabular data.
+```javascript
+import * as React from 'react';
+import {Table} from 'baseui/table';
+const data = {
```

Comment:
I take it this is supposed to be an array of objects? And should it not be an [[], []] for rows?

Reply:
I think your suggestion of using the nested array data structure  `[[],[]]` is another possibility and is commonly seen for handling csv data where the first row is a row of headers and the following rows are rows of text data.

An example would be:
```
[
  [ 'Emp_ID', 'First', 'Last']
  [1, 'John', 'Smith']
  [2, 'Jane', 'Doe']
]
```

The design I have suggested is inspired by ant design's table https://ant.design/components/table/#How-To-Use and one of our projects also has a similar implementation

Reply:
Gotcha yeah I see, but this should still be 
```
[{}, {}]
```

Reply:
we landed on tabular format